NPE in XtextPluginImages when initialized from non-UI thread

A `NullPointerException` occurrs when `XtextPluginImages` is lazy-initialized from a non-UI thread.

Steps to reproduce:
1. Enable *General|Editors|Text Editors|Show code minings for problem annotations*
2. Make sure there is an appropriate problem in an Xtext-related file, such as an Xtend file
3. Close all editors and restart Eclipse
4. Double-click the problem marker in the *Problems* view, causing the editor to open

Expected behavior:
The editor opens without incident.

Current behavior:
```
java.util.concurrent.CompletionException: java.lang.NullPointerException: toQuery
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.lang.NullPointerException: toQuery
	at java.base/java.util.Objects.requireNonNull(Objects.java:259)
	at org.eclipse.jface.resource.JFaceResources.getResources(JFaceResources.java:213)
	at org.eclipse.jface.resource.ImageRegistry.<init>(ImageRegistry.java:150)
	at org.eclipse.jface.resource.ImageRegistry.<init>(ImageRegistry.java:122)
	at org.eclipse.xtext.ui.internal.XtextPluginImages.initialize(XtextPluginImages.java:179)
	at org.eclipse.xtext.ui.internal.XtextPluginImages.ensureInitialized(XtextPluginImages.java:143)
	at org.eclipse.xtext.ui.internal.XtextPluginImages.getAnnotationImagesNonfixable(XtextPluginImages.java:203)
	at org.eclipse.xtext.ui.editor.model.XtextMarkerAnnotationImageProvider.getImages(XtextMarkerAnnotationImageProvider.java:51)
	at org.eclipse.xtext.ui.editor.model.XtextMarkerAnnotationImageProvider.getManagedImage(XtextMarkerAnnotationImageProvider.java:39)
	at org.eclipse.ui.texteditor.DefaultMarkerAnnotationAccess.getImage(DefaultMarkerAnnotationAccess.java:371)
	at org.eclipse.ui.texteditor.DefaultMarkerAnnotationAccess.isPaintable(DefaultMarkerAnnotationAccess.java:298)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningFilter.isPaintable(AnnotationCodeMiningFilter.java:101)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1932)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningProvider.createCodeMinings(AnnotationCodeMiningProvider.java:270)
	at org.eclipse.ui.internal.editors.text.codemining.annotation.AnnotationCodeMiningProvider.lambda$0(AnnotationCodeMiningProvider.java:219)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	... 6 more
```

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
@@ -29,7 +29,7 @@ import org.eclipse.xtext.ui.editor.XtextEditor;
  * PDEPluginImages.
  * 
  * @author Peter Friese - Initial contribution and API
- * @author Dennis Hübner
+ * @author Dennis HÃ¼bner
  * 
  */
 public class XtextPluginImages {
@@ -199,6 +199,7 @@ public class XtextPluginImages {
 	}
 
 	public static Image manage(String key, ImageDescriptor desc) {
+		ensureInitialized();
 		if (PLUGIN_REGISTRY == null)
 			return null;
 		Image image = desc.createImage();

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
@@ -135,6 +135,8 @@ public class XtextPluginImages {
 
 	public static Image get(String key) {
 		ensureInitialized();
+		if (PLUGIN_REGISTRY == null)
+			return null;
 		return PLUGIN_REGISTRY.get(key);
 	}
 
@@ -197,6 +199,8 @@ public class XtextPluginImages {
 	}
 
 	public static Image manage(String key, ImageDescriptor desc) {
+		if (PLUGIN_REGISTRY == null)
+			return null;
 		Image image = desc.createImage();
 		PLUGIN_REGISTRY.put(key, image);
 		return image;

--- a/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
+++ b/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/internal/XtextPluginImages.java
@@ -139,7 +139,10 @@ public class XtextPluginImages {
 	}
 
 	private static void ensureInitialized() {
-		if (PLUGIN_REGISTRY == null)
+		if (PLUGIN_REGISTRY == null &&
+				// to avoid NPE in XtextPluginImages when initialized from non-UI thread
+				// https://github.com/eclipse/xtext/issues/2807
+				Display.getCurrent() != null)
 			initialize();
 	}
 	

